pkg/nsx/services/subnetport/builder.go

```diff
@@ -0,0 +1,90 @@
+package subnetport
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+
+	"github.com/google/uuid"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+
+	"github.com/vmware/vsphere-automation-sdk-go/services/nsxt/model"
+
+	"github.com/vmware-tanzu/nsx-operator/pkg/apis/v1alpha1"
+	"github.com/vmware-tanzu/nsx-operator/pkg/nsx/services/common"
+)
+
+var (
+	String = common.String
+)
+
+func (service *SubnetPortService) buildSubnetPort(obj *v1alpha1.SubnetPort) (*model.SegmentPort, error) {
+	allocateAddresses := "BOTH"
+	nsxSubnetPortName := fmt.Sprintf("port-%s", obj.Name)
+	nsxSubnetPortID := string(obj.UID)
+	// use the subnetPort CR UID as the attachment uid generation to ensure the latter stable
+	nsxCIFID, err := uuid.NewRandomFromReader(bytes.NewReader([]byte(nsxSubnetPortID)))
+	if err != nil {
+		return nil, err
+	}
+	// TODO: Get the subnet's NSX resource path from subnet store after it is implemented.
+	nsxSubnetPath, err := service.GetSubnetPathForSubnetPort(obj)
+	if err != nil {
+		return nil, err
+	}
+	namespace := &corev1.Namespace{}
+	namespacedName := types.NamespacedName{
+		Name: obj.Namespace,
+	}
+	if err := service.Client.Get(context.Background(), namespacedName, namespace); err != nil {
+		return nil, err
+	}
+	namespace_uid := namespace.UID
+	// TODO: set AppId and ContextId for pod port.
+	nsxSubnetPort := &model.SegmentPort{
+		DisplayName: common.String(nsxSubnetPortName),
+		Id:          common.String(nsxSubnetPortID),
+		Attachment: &model.PortAttachment{
+			AllocateAddresses: &allocateAddresses,
+			// AppId:             common.String("nsx.ns-3.pod-5%"),
+			// ContextId:         common.String("95ccaad4-0dfb-469e-a1e6-27d815826382"),
+			Id:         common.String(nsxCIFID.String()),
+			TrafficTag: nil,
+			Type_:      common.String("STATIC"),
+		},
+		Tags: service.buildBasicTags(obj, namespace_uid),
+		Path: &nsxSubnetPath,
```

Comment:
do we need to set the nsxSubnetPath here?

Reply:
We need the path to check the realized state (I write it here so that I don't need to create another request to get the resource from NSX). btw the path should be `&nsxSubnetPortPath` instead of `&nsxSubnetPath`, I just fixed it. Thanks for your comment.